apps/passport-client/src/appHooks.ts

```diff
@@ -395,3 +395,53 @@ export const useIOSOrientationFix = (): void => {
     };
   }, []);
 };
+
+/**
+ * Hook that attempts to login from the one_click_redirect key in local storage.
+ * This is set inside the static one click login page.
+ * If this key exists it will attempt to login with the one-click-login flow
+ * even if a different user is logged in.
+ */
+const ONE_CLICK_REDIRECT_KEY = "one_click_redirect";
+export const useAutoLoginFromOneClick = (): { loading: boolean } => {
+  const dispatch = useDispatch();
+  const self = useSelf();
+  const attemptedLogin = useRef(false);
+  const [oneClickRedirect, setOneClickRedirect] = useState(
+    localStorage.getItem(ONE_CLICK_REDIRECT_KEY)
+  );
+
+  useEffect(() => {
+    if (attemptedLogin.current) return;
+    // prevent multiple login attempts
+    attemptedLogin.current = true;
+    if (!oneClickRedirect) return;
+
+    const attemptAutoLogin = async (): Promise<void> => {
+      try {
+        const [, , email, code] = oneClickRedirect.split("/");
+
+        // If the same user is already logged in we don't want to auto-login again
+        if (self?.emails?.includes(email))
+          throw new Error("User is already logged in");
+
+        // Attempt to login with the one-click-login flow
+        await dispatch({
+          type: "one-click-login",
+          email,
+          code,
+          targetFolder: undefined
+        });
```

Comment:
This will happen even if the user is logged in @rrrliu 
Is this ok?

Reply:
let's do the redirect approach of /#/one-click-preview for now, that way we build off of the previous logic